docs: Update 0.18.x release notes (#5382)
hashicorp/boundary: website/content/docs/release-notes/v0_18_0.mdx

    Learn more:&nbsp; <a href="#known-issues-and-breaking-changes">Known issues and breaking changes </a>
    </td>
  </tr>

<tr>
    <td style={{verticalAlign: 'middle'}}>

>     Go version 1.23.3 x509 key pair behavior changes

Reviewer: I believe 18.0 uses go 1.23.1 (according to go.mod) and the behaviour was introduced with Go 1.23, so it will have been present since then.

Author: Thank you for the clarification! I changed the references to Boundary controllers/workers to `0.18.x` and the Go version to `1.23`.

    </td>
    <td style={{verticalAlign: 'middle'}}>

>     Boundary version 0.18.x uses Go version 1.23, which introduced a new x509 key pair behavior. Some VPN implementations struggle with the TLS handshake being sent over 2 frames instead of 1, which can lead to Boundary version 0.18.x controllers or workers being unable to establish SSH connections. As a workaround, you can revert back to the previous key pair behavior.

Reviewer 2: ```suggestion
    Boundary version 0.18.x uses Go version 1.23, which introduced a new TLS handshake behavior. Some VPN providers struggle with the TLS handshake being sent over 2 frames instead of 1, which can lead to Boundary version 0.18.x controllers, workers or clients being unable to establish connections. As a workaround, you can revert back to the previous TLS handshake behavior.
```

The x509 key pair stuff is probably just going to confuse customers, we can just talk about the TLS handshake.

    </td>
  </tr>

  <tr>
    <td style={{verticalAlign: 'middle'}}>
    0.18.x
    </td>
    <td style={{verticalAlign: 'middle'}}>

>     Boundary version 0.18.x controllers or workers are unable to establish SSH connections using the <code>boundary connect ssh</code> command

Reviewer 2: I think we've only had reports of this happening with the CLI, but in theory it could happen on any VPN network boundary.
```suggestion
    Boundary version 0.18.x CLI is unable to establish connections using the <code>boundary connect</code> command.
```

    </td>
    <td style={{verticalAlign: 'middle'}}>

>     Boundary version 0.18.x uses Go version 1.23, which introduced a new x509 key pair behavior. Some VPN implementations struggle with the TLS handshake being sent over 2 frames instead of 1, which can lead to Boundary version 0.18.x controllers or workers being unable to establish SSH connections.

Reviewer 2: ```suggestion
    Boundary version 0.18.x uses Go version 1.23, which introduced a new TLS handshake behavior. Some VPN providers struggle with the TLS handshake being sent over 2 frames instead of 1, which can lead to Boundary version 0.18.x controllers, workers or clients being unable to establish connections. As a workaround, you can revert back to the previous TLS handshake behavior.
```

    <br /><br />

>     As a workaround, you can revert back to the previous key pair behavior by adding the <code>tlskyber=0</code> and <code>x509keypairleaf=0</code> parameters to the GODEBUG environment variable before the <code>boundary connect ssh command</code>. For example:

Reviewer 2: Only the TLS kyber variable was necessary
```suggestion
    As a workaround, you can revert back to the previous TLS handshake behavior by adding the <code>tlskyber=0</code> parameters to the GODEBUG environment variable before the <code>boundary connect</code> command. For example:
```

    <br /><br />

>     <code>GODEBUG=tlskyber=0,x509keypairleaf=0 boundary connect ssh -target-id&lt;ID&gt;</code>

Reviewer 2: ```suggestion
    <code>GODEBUG=tlskyber=0 boundary connect ssh -target-id &lt;ID&gt;</code>
```

    <br /><br />

>     Learn more: <a href="https://tip.golang.org/doc/go1.23">Go 1.23 Release Notes</a>

Reviewer 2: I think we should add a link to the issue where this is discussed, which will help users figure out that actually this is a VPN problem, not a Boundary problem: https://github.com/golang/go/issues/70047

Author: Sounds good! I added the link to the issue in addition to the release notes, but I can also remove the release notes link if you don't think they're relevant.
